### PR TITLE
Kmp 2to3 fixes poetry (C4-1106)

### PR DESCRIPTION
**Pull request by netsettler**
This adds poetry, PEP8, and some fixes to the tests and the repo.

### CODE REVIEW

File: .github/workflows/main-publish.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
```
Comment (willronchetti) [inline .github/workflows/main-publish.yml]: 22.04?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    name: Python ${{ matrix.python_version }} Tests
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
```
Comment (willronchetti) [inline .github/workflows/main.yml]: 22.04?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    strategy:
+      matrix:
+        python_version: ['3.7', '3.8', '3.9', '3.10', '3.11']
```
Comment (willronchetti) [inline .github/workflows/main.yml]: No need for 3.7 I'd say

File: setup.py [hunk 4]
```diff
@@ -1,26 +1,31 @@
+import glob
```
Comment (willronchetti) [inline setup.py]: This file can be deleted entirely no?

File: version.py [hunk 5]
```diff
@@ -0,0 +1 @@
+vcf/__init__.py
```
Comment (willronchetti) [inline version.py]: Can be deleted no?

**Review by willronchetti — APPROVED** [review]
Looks good should be tested in CGAP though

File: setup.py (hunk 4 above)
Comment (netsettler) [reply setup.py]: Seemingly. It took some adjusting but seems to be working now.

File: version.py (hunk 5 above)
Comment (netsettler) [reply version.py]: Had to reimplement how it `__init__.py` gets the version, but yeah, I don't think `version.py` is doing anything.

Comment (netsettler) [reply version.py]: In my revised version, it gets the VERSION from `pyproject.toml` (indirectly through `importlib.metadata`) whereas previously `__init__.py` was the source of authority on version.

File: .github/workflows/main-publish.yml (hunk 1 above)
Comment (netsettler) [reply .github/workflows/main-publish.yml]: Thanks. Fixed.

File: .github/workflows/main.yml (hunk 2 above)
Comment (netsettler) [reply .github/workflows/main.yml]: Fixed, too.

**Comment by netsettler** [comment]
This tested cleanly in cgap-portal testing for Python 3.8 on my local machine.
```
poetry run python -m pytest -vv -m static && make lint
===== 3 passed, 2175 deselected, 238 warnings in 8.75s =====
poetry run python -m pytest -xvv -r w --durations=25 --timeout=600 -m "not manual and not sloppy and not static and not broken and not performance and not integratedx and not indexing"
===== 1087 passed, 15 skipped, 1076 deselected, 944 warnings in 751.99s (0:12:31) =====
poetry run python -m pytest -xvv -r w --durations=25 --timeout=200 -m "not manual and not sloppy and not static and not broken and not performance and not integratedx and indexing and es"
===== 20 passed, 2158 deselected, 647 warnings in 177.56s (0:02:57) =====
poetry run python -m pytest -xvv -r w --durations=25 --timeout=200 -m "not manual and not sloppy and not static and not broken and not performance and not integratedx and indexing and not es"
===== 616 passed, 72 skipped, 1490 deselected, 1609 warnings in 121.91s (0:02:01) =====
commit 0467c8be1fd8b8a86ada9d6d110e436f424ab039 (HEAD -> upgrade-python-20230925, origin/upgrade-python-20230925)
Wed Oct  4 19:00:34 EDT 2023
```
Also [in Python 3.11 on GA](https://github.com/dbmi-bgm/cgap-portal/actions/runs/6412636688?pr=756).

**Comment by netsettler** [comment]
OK, I've moved the label to 3.0.0 since, by agreement with @willronchetti on Slack, passing tests in `cgap-portal` is a good test. No need for test deploy, he also agreed. (And, as a contingency in case of later problems, if anyone needs to revert, they can always use `^2.0.0` and they won't end up in the `3.0.0` series.) I think this is ready to merge.

**Pull request merged**